src/bijectors/leaky_relu.jl

```diff
@@ -1,101 +1,29 @@
-    function reconstruct_leakyrelu(xs)
-        return LeakyReLU{typeof(xs.α),N}(xs.α)
-    end
-    return (α = x.α,), reconstruct_leakyrelu
-end
-
-up1(b::LeakyReLU{T, N}) where {T, N} = LeakyReLU{T, N + 1}(b.α)
-
-# (N=0) Univariate case
-function (b::LeakyReLU{<:Any, 0})(x::Real)
-    mask = x < zero(x)
-    return mask * b.α * x + !mask * x
-end
-(b::LeakyReLU{<:Any, 0})(x::AbstractVector{<:Real}) = map(b, x)
-
-function inverse(b::LeakyReLU{<:Any,N}) where N
-    invα = inv.(b.α)
-    return LeakyReLU{typeof(invα),N}(invα)
-end
-
-function logabsdetjac(b::LeakyReLU{<:Any, 0}, x::Real)
-    mask = x < zero(x)
-    J = mask * b.α + (1 - mask) * one(x)
-    return log(abs(J))
-end
-logabsdetjac(b::LeakyReLU{<:Real, 0}, x::AbstractVector{<:Real}) = map(x -> logabsdetjac(b, x), x)
+inverse(b::LeakyReLU) = LeakyReLU(inv.(b.α))
 
-
-# We implement `with_logabsdet_jacobian` by hand since we can re-use the computation of
-# the Jacobian of the transformation. This will lead to faster sampling
-# when using `rand` on a `TransformedDistribution` making use of `LeakyReLU`.
-function with_logabsdet_jacobian(b::LeakyReLU{<:Any, 0}, x::Real)
+function with_logabsdet_jacobian(b::LeakyReLU, x::Real)
     mask = x < zero(x)
-    J = mask * b.α + !mask * one(x)
-    return (J * x, log(abs(J)))
-end
-
-# Batched version
-function with_logabsdet_jacobian(b::LeakyReLU{<:Any, 0}, x::AbstractVector)
-    J = let T = eltype(x), z = zero(T), o = one(T)
-        @. (x < z) * b.α + (x > z) * o
-    end
-    return (J .* x, log.(abs.(J)))
-end
-
-# (N=1) Multivariate case
-function (b::LeakyReLU{<:Any, 1})(x::AbstractVecOrMat)
-    return let z = zero(eltype(x))
-        @. (x < z) * b.α * x + (x > z) * x
-    end
+    J = mask * b.α + !mask
+    return J * x, log(abs(J))
 end
 
-function logabsdetjac(b::LeakyReLU{<:Any, 1}, x::AbstractVecOrMat)
-    # Is really diagonal of jacobian
-    J = let T = eltype(x), z = zero(T), o = one(T)
-        @. (x < z) * b.α + (x > z) * o
-    end
-
-    if x isa AbstractVector
-        return sum(log.(abs.(J)))
-    elseif x isa AbstractMatrix
-        return vec(sum(log.(abs.(J)); dims = 1))  # sum along column
-    end
-end
-
-# We implement `forward` by hand since we can re-use the computation of
-# the Jacobian of the transformation. This will lead to faster sampling
-# when using `rand` on a `TransformedDistribution` making use of `LeakyReLU`.
-function with_logabsdet_jacobian(b::LeakyReLU{<:Any, 1}, x::AbstractVecOrMat)
-    # Is really diagonal of jacobian
-    J = let T = eltype(x), z = zero(T), o = one(T)
-        @. (x < z) * b.α + (x > z) * o
-    end
-
-    if x isa AbstractVector
-        logjac = sum(log.(abs.(J)))
-    elseif x isa AbstractMatrix
-        logjac = vec(sum(log.(abs.(J)); dims = 1))  # sum along column
-    end
-
-    y = J .* x
-    return (y, logjac)
+# Array inputs.
+function with_logabsdet_jacobian(b::LeakyReLU, x::AbstractArray)
+    mask = x .< zero(eltype(x))
+    J = mask .* b.α .+ (!).(mask)
```

Comment:
Just something like
```suggestion
    J = broadcast(b.α, x) do α, xi
        return x < zero(x) ? α : one(α)
    end
```
Could possibly even factor out the scalar method and re-use it here?

Reply:
See above comment